### PR TITLE
Use peep instead of pip (bug 1118379)

### PR DESCRIPTION
**Pull request by magopian**
Fixes [bug 1118379](https://bugzilla.mozilla.org/show_bug.cgi?id=1118379)

Use https://pypi.python.org/pypi/peep instead of plain pip

### CODE REVIEW

**Comment by magopian** [comment]
@andymckay r?

**Comment by robhudson** [comment]
I think this should also copy the peep.py file to the repo and verify it, otherwise peep itself is the weakest link in the chain. See the peep README for this suggestion.

**Comment by andymckay** [comment]
I was going to suggest install peep off of pyrepo and then you don't need to install the rest of pyrepo. Embedding it works too.

File: docs/topics/install-olympia/packages.rst [hunk 1]
```diff
@@ -32,3 +32,16 @@ Or more simply with::
...
 The latter will also install the npm dependencies.
+
+
+Using peep instead of pip
+-------------------------
+
+To have reproducible environments and make sure what we install is what we
+wanted to install, we're moving to use peep_ instead of pip::
+
+    pip install peep
```
Comment (andymckay) [inline docs/topics/install-olympia/packages.rst]: This should be pulling off pyrepo with == to specify the version, or embed it.

**Comment by magopian** [comment]
This is currently broken, I believe because of the incompatibility of peep and pip7 (eg https://github.com/erikrose/peep/pull/94)

Also, I'm not sure we really need/want to use peep. It would allow us to not need pyrepo anymore, installing straight from pypi. However, we'd then have other issues (what if a package is badly updated on pypi as it happened for the python-memcached 1.53 wheel which is in fact now a 1.55 package? Or if someone removes an old version or deletes a package from pypi?).

So I believe we need to keep pyrepo, and we thus don't need peep. Thoughts @andymckay @robhudson?

**Comment by kmaglione** [comment]
I think I'd still rather have peep, even if we stick with pyrepo. We thought we were getting all of our packages from our wheelhouse before, and things still broke, and were very difficult to diagnose. If we can manage to get this working without making our lives too difficult in the future, I think it's worth it.

**Comment by andymckay** [comment]
I agree with @kmaglione, peep is worth it.

**Comment by magopian** [comment]
Closing for now: the [peep bug](https://github.com/erikrose/peep/pull/94) is still open.

If this is reopened, it'll also need some more documentation around how to maintain the requirements files (updating, adding a new requirement, adding the appropriate hashes). Might need a lot of rework if https://github.com/mozilla/olympia/pull/795 makes it to master (it got reverted in https://github.com/mozilla/olympia/commit/fa65808a2d5e009f6ffdd7de10df989307928360)

**Pull request closed without merging**